src/metadata.ts

```diff
@@ -9,42 +11,73 @@ export interface IPageMetadata {
-Compete against your online friends or play locally. Free and open-source software project.`,
-  noindex: true,
-};
+function getDefaultMetadata(lang: string): IPageMetadata {
+  return {
+    title: trans('metadata.default.title', lang),
+    description: trans('metadata.default.description', lang),
+    noindex: true,
+  };
+}
 
 // Most specific URLs MUST come first.
 const PAGES_METADATA: IPageMetadata[] = [
   {
-    title: TITLE_PREFIX + 'About Us',
-    description: 'About FreeBoardGame.org, a free and open-source software project.',
+    title: TITLE_PREFIX + 'metadata.about.title',
+    description: 'metadata.about.description',
     url: new RegExp('^/about', 'i'),
   },
   {
-    title: TITLE_PREFIX + 'Play Free Board Games Online',
-    description: `Play board games in your browser for free. \
-Compete against your online friends or play locally. Free and open-source software project.`,
+    title: TITLE_PREFIX + 'metadata.about.title',
+    description: 'metadata.index.description',
     url: new RegExp('^/$', 'i'),
   },
 ];
 
-function getGamesPageMetadata(): IPageMetadata[] {
-  return GAMES_LIST.map(gameDef => ({
-    title: TITLE_PREFIX + `Play Free ${gameDef.name} Online`,
-    description: gameDef.descriptionTag,
-    url: new RegExp(`^/g/${gameDef.code}$`, 'i'),
-  }));
+function getAllPagesMetadata(lang: string): IPageMetadata[] {
+  const otherPagesMetadata = PAGES_METADATA.map(page => {
+    return { ...page, title: trans(page.title, lang), description: trans(page.description, lang) };
+  });
+  const gamePagesMetadata = getGamesPageMetadata(lang);
+  const allPagesMetadata: IPageMetadata[] = [...otherPagesMetadata, ...gamePagesMetadata];
+  return allPagesMetadata;
+}
+
+function getGamesPageMetadata(lang: string): IPageMetadata[] {
+  return GAMES_LIST.map(gameDef => {
+    const titleTransTemplate = trans('metadata.gameTemplate.title', lang);
+    const titleTranslated = titleTransTemplate.replace('${GAME}', gameDef.name);
+    return {
+      title: TITLE_PREFIX + titleTranslated,
+      description: trans(gameDef.descriptionTag, lang),
+      url: new RegExp(`^/g/${gameDef.code}$`, 'i'),
+    };
+  });
+}
+
+export function getLangFromURL(url: string, translationsDict: any) {
+  const match = url.match(/\/([A-Za-z]{2})/);
+  if (match) {
+    const lang = match[1].toLowerCase();
+    if (lang in translationsDict) {
+      return lang;
+    }
+  }
 }
 
 /** Given a URL, gets its title. */
 export const getPageMetadata = (url: string): IPageMetadata => {
-  const allPagesMetadata = [...PAGES_METADATA, ...getGamesPageMetadata()];
-  const metadata = allPagesMetadata.find(meta => meta.url!.test(url));
+  let lang = getLangFromURL(url, translations);
+  if (!lang) {
+    // fallback to English
+    lang = 'en';
+  }
+  registerLang(lang, translations[lang]);
+  const allPagesMetadata = getAllPagesMetadata(lang);
+  let metadata = allPagesMetadata.find(meta => meta.url!.test(url));
```

Comment:
This seem to be very wasteful. You want only one page and you are generating the IPageMetadata of everyone. if you execute this to all pages, it will be O(n^2) instead of O(n). 

You could instead change getAllPagesMetadata and getGamesPageMetadata to only be for one page at a time.